src/Workspaces/Remote/ServiceHub/Services/ProcessTelemetry/RemoteProcessTelemetryService.PerformanceReporter.cs

```diff
@@ -63,38 +64,43 @@ protected override void OnPaused()
                         {
-                            // this will report telemetry under VS. this will let us see how accurate our performance tracking is
-                            RoslynLogger.Log(FunctionId.Diagnostics_BadAnalyzer, KeyValueLogMessage.Create(m =>
-                            {
-                                // since it is telemetry, we hash analyzer name if it is not builtin analyzer
-                                m[nameof(analyzerInfo.AnalyzerId)] = isInternalUser ? analyzerInfo.AnalyzerId : analyzerInfo.PIISafeAnalyzerId;
-                                m[nameof(analyzerInfo.LocalOutlierFactor)] = analyzerInfo.LocalOutlierFactor;
-                                m[nameof(analyzerInfo.Average)] = analyzerInfo.Average;
-                                m[nameof(analyzerInfo.AdjustedStandardDeviation)] = analyzerInfo.AdjustedStandardDeviation;
-                            }));
-                        }
+                            var newAnalyzer = reported.Add(analyzerInfo.AnalyzerId);
 
-                        // for logging, we only log once. we log here so that we can ask users to provide this log to us
-                        // when we want to find out VS performance issue that could be caused by analyzer
-                        if (newAnalyzer)
-                        {
-                            _logger.TraceEvent(TraceEventType.Warning, 0,
-                                $"Analyzer perf indicators exceeded threshold for '{analyzerInfo.AnalyzerId}' ({analyzerInfo.AnalyzerIdHash}): " +
-                                $"LOF: {analyzerInfo.LocalOutlierFactor}, Avg: {analyzerInfo.Average}, Stddev: {analyzerInfo.AdjustedStandardDeviation}");
+                            var isInternalUser = _telemetrySession.IsUserMicrosoftInternal;
+
+                            // we only report same analyzer once unless it is internal user
+                            if (isInternalUser || newAnalyzer)
+                            {
+                                // this will report telemetry under VS. this will let us see how accurate our performance tracking is
+                                RoslynLogger.Log(FunctionId.Diagnostics_BadAnalyzer, KeyValueLogMessage.Create(m =>
+                                {
+                                    // since it is telemetry, we hash analyzer name if it is not builtin analyzer
+                                    m[nameof(analyzerInfo.AnalyzerId)] = isInternalUser ? analyzerInfo.AnalyzerId : analyzerInfo.PIISafeAnalyzerId;
+                                    m[nameof(analyzerInfo.LocalOutlierFactor)] = analyzerInfo.LocalOutlierFactor;
+                                    m[nameof(analyzerInfo.Average)] = analyzerInfo.Average;
+                                    m[nameof(analyzerInfo.AdjustedStandardDeviation)] = analyzerInfo.AdjustedStandardDeviation;
+                                    m[nameof(forSpanAnalysis)] = forSpanAnalysis;
```

Comment:
@jmarolf This is the primary differentiator for bad analyzer telemetry for lightbulb vs full document analysis. Hope this suffices for the kusto queries.